### ISSUE TITLE
Cursor movement bug on Android Devices

### ISSUE THREAD

**Issue by DiMoNTD** [igorescobar/jQuery-Mask-Plugin#432]
@igorescobar, have some bug with input mask on Android devices. When typing digits in "Telephone" input, cursor stay still and values are added immediately after it.

For better understanding, I've recorded the video on device:
https://www.dropbox.com/s/0igokg7vvx8l0us/2016_07_18_13_12_37.mp4?dl=0

Samsung Galaxy Note 3
Android: 5.0
Chrome version: 51.0.2704.81
Plugin version: 1.14.0

**UPDATE:** after little research, I've founded closed issue #348 and fix in version [1.13.5](https://github.com/igorescobar/jQuery-Mask-Plugin/releases/tag/v1.13.5). I've installed this version in my project and tested with same Andoid device, and it's work fine. With lastest version [1.14.0](https://github.com/igorescobar/jQuery-Mask-Plugin/releases/tag/v1.14.0) bug is reproduced

**Comment by clifforama** [comment]
I notice the same. It's OK in 1.13.9, but broken again in 1.14.0.

**Comment by ThatMouse** [comment]
I can confirm cursor issue is in 1.14.0, but my sample works perfectly in 1.13.9. I have the same issue using RobinHerbots's inputmask. Cursor/Caret remains on left side of textbox and text does not input correctly.

Android 6.0.1 Nexus 5 / Chrome 52.0.2743.98

Not the best sample, but here's what I was testing with. I just need something that only allows alpha and some other characters that are common in people's names:
$("#tbFirstName").mask("BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB", { placeholder: "______________________", translation: { 'B': { pattern: /[a-zA-Z\-., ]+/ } } });

**Comment by pmccloghrylaing** [comment]
It seems to be related to predictive text in Android (https://github.com/effektif/react-mentions/issues/30). Depending on the situation "type=tel" or "autocomplete=off" are possible workarounds. Looks like they did some predictive text detection to fix the problem.

**Comment by igorescobar** [comment]
Does anyone knows what I should do to fix this? LOL

**Comment by diebugger** [comment]
Hi @igorescobar, 
I tried to recover the function "setCaret" from version 1.13.9 and put it into the 1.14.0, in order to save the useful cleanVal() and masked() functions... and it seems to work on my Android device and on Chrome web browser. Hope this hint will help.
Keep on the good work!

**Comment by igorescobar** [comment]
@diebugger Could you submit it as a PR?

**Comment by igorescobar** [comment]
@DiMoNTD @clifforama @ThatMouse @pmccloghrylaing @diebugger can you guys test this PR? https://github.com/igorescobar/jQuery-Mask-Plugin/pull/464

**Comment by pmccloghrylaing** [comment]
I've tested it and it causes a different problem.
I tested it this way:
- '0000-0000' as a mask on a standard `type="text"` input
- Type in '12345678'
- Move the caret to before the '5', directly after the '-'
- Type '123'
- Expected result is '1234-1235'

The current version of the plugin produces '1234-3215' as the caret stays just after the '-'. This pull request produces '1234-1567' because the caret gets moved to the end.
